### DISCUSSION TITLE
enzyme/BUILD file conflicts with CMake convention, IDE integration

### DISCUSSION THREAD

**Discussion by samuelpmish** [EnzymeAD/Enzyme#1790] [General]
A common CMake convention is to name the build directory simply "build", but on my mac this conflicts with the BUILD file in enzyme. As a result, the instructions on the main readme don't work

```zsh
% cd Enzyme/enzyme 
% cmake . -Bbuild 
CMake Error: Unable to (re)create the private pkgRedirects directory:
/Users/sam/code/Enzyme/enzyme/build/CMakeFiles/pkgRedirects
```

and CMake emits a cryptic error message (the reason it can't make that directory is because `build` is a file). Is it possible to rename that BUILD file to something else to avoid this conflict with case insensitive file systems?

**Comment by wsmoses** [comment]
So unfortunately that is the required file for the bazel build system.

**Reply by samuelpmish** [reply]
ah, I'm not familiar with bazel. Maybe we can update the readme to suggest a different build directory name for cmake then
